### DISCUSSION TITLE
Appropriate Section Numbering

### DISCUSSION THREAD

**Discussion by sammo3182** [quarto-dev/quarto-cli#1748] [Q&A]
I was excited about the [custom setting of references](https://quarto.org/docs/authoring/cross-references.html#references-1) but cannot find the correct way to set them. In the example below, I tried to change the section and figure labels to alphabetic and sub labels to roman. But as the screenshot shows, none of the section numbers, figure numbers, or their references changed in either pdf or HTML format. I tried to move `number-sections: true` and/or `crossref` under `pdf`. Nothing changed.  

What I hope to get are:

1. Customizing the section, table, and figure numbering and correctly cross-referring them. (#873 discussed this issue, but I wonder if there is an easier way to do this for HTML, pdf, and docx all)
2. Setting the numbering of some sections differently from the previous, for instance, `arabic` for main content sections and `alpha` for appendices. (#1283 post a similar demand but it seems no good solution yet)

````markdown
---
title: "test"

number-sections: true

format: 
  pdf: default
  html: default
  
crossref:
  sec-labels: alpha A
  subref-labels: roman i
---

# section 

@fig-test

# section {#sec-test1}
## subsection {#sec-test2}

@sec-test1

@sec-test2

# section 

# References {.unnumbered}

```{r fig-test}
#| fig-cap: "A plot"

plot(mtcars$mpg)
```

````

![image](https://user-images.githubusercontent.com/6463211/183531672-700941fd-f970-4e30-a8e7-dbad1116d677.png)

**Comment by mcanouil** [comment]
you can't mix the type of numbering within the same document, because the type is defined at the top-level of the document, meaning you have to hack your way to what you want using a Lua filter for example.

`sec-labels` does not exist if I am not mistaken. It would be `labels` for the default labels numbering and then you can change `fig-labels`, `tbl-labels`, and `subref-labels`.
In your example, you only have figures and not tables nor sub figures, thus your options have no effect.
See https://quarto.org/docs/authoring/cross-references.html#numbering
i.e. for your example:
```yml
crossref:
  labels: alpha A
  fig-labels: roman i
```

**Reply by sammo3182** [reply]
Hi Michael @mcanouil, thank you for responding to the question, but the codes you gave I've tried and do not work. If you click the link I gave in the initial post, you'll find we share the same source. If the codes work for you, a replicable example will be super appreciated. Btw, I didn't make up the `sec-labels`, it was given by the Rstudio autocomplition, which I guess can be traced back to the parameter setting, although I'm not exactly sure.

**Reply by mcanouil** [reply]
I tried it and found out the `sec-labels` (not documented).
Section numbering does not change, which seems to be a bug.
But `fig-labels` works as expected for HTML format.
`crossref` has no effect for pdf format.

- `html`
  <img width="1624" alt="screenshot of VSCode showing a quarto document with crossref labels numbering changed on the left and the rendered html showing no changes for section numbering but correct changes for figures numbering" src="https://user-images.githubusercontent.com/8896044/183546578-e91b293c-b4cf-4ccd-a83c-24e40d24f057.png">

- `pdf`
  <img width="1624" alt="screenshot of VSCode showing a quarto document with crossref labels numbering changed on the left and the rendered pdf showing no changes for all modified numbering crossrefs" src="https://user-images.githubusercontent.com/8896044/183546811-e4b5a038-5fb8-4634-ba82-450923efe77a.png">

**Reply by mcanouil** [reply]
I would open an issue from this discussion (yellow square from the screenshot below) or wait for the maintainers to confirm it is indeed a bug.
![image](https://user-images.githubusercontent.com/8896044/183547347-26b277d1-16e3-47eb-a805-a9e6066cb097.png)
